src/bp/ui-admin/src/telemetry/client.ts

```diff
@@ -0,0 +1,143 @@
+import axios from 'axios'
+import { lang } from 'botpress/shared'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import uuid from 'uuid'
+
+import store from '../store'
+
+interface EventPackageInfoType {
+  [key: string]: {
+    locked: boolean
+    timeout: string
+    getPackage: Function
+  }
+}
+
+export const axiosConfig = {
+  baseURL: window.TELEMETRY_URL,
+  headers: {
+    withCredentials: false
+  }
+}
+
+const dataSchemaVersion = '1.0.0'
+
+const telemetrySchemaVersion = '1.0.0'
+
+const storeInfos: Function[] = []
+
+const eventPackageInfo: EventPackageInfoType = {}
+
+const toggleLock = (eventName: string) => {
+  if (_.has(eventPackageInfo, eventName)) {
+    eventPackageInfo[eventName].locked = !eventPackageInfo[eventName].locked
+  }
+}
+
+const getEventLockTimeout = (event: string) => {
+  const item = window.localStorage.getItem(event)
+  if (item !== null) {
+    const timeout = parseInt(item) - new Date().getTime()
+    if (timeout > 0) {
+      return timeout
+    }
+    window.localStorage.removeItem(event)
+  }
+  return 0
+}
+
+const addEventLockTimeout = (event: string, timeout: number) => {
+  setTimeout(() => toggleLock(event), timeout)
+  window.localStorage.setItem(event, (timeout + moment().valueOf()).toString())
+}
+
+const checkStoreInfoReceived = () => !storeInfos.find(info => info() === undefined)
+
+export const trackReduxStoreInfo = (pathInStore: string) => {
+  storeInfos.push(() => {
+    return _.get(store.getState(), pathInStore)
+  })
+}
```

Comment:
These mini methods are so simple that they just complicate understanding the underlying code. I have no clue what's going on

Reply:
Fixed in : 
52d479c